echopype/echodata/echodata.py

```diff
@@ -133,6 +142,118 @@ def __read_converted(self, converted_raw_path: Optional["PathHint"]):
 
         self.converted_raw_path = converted_raw_path
 
+    def _set_tree(self, tree: DataTree):
+        self._tree = tree
+
+    @classmethod
+    def from_file(cls, converted_raw_path, storage_options=None, open_kwargs={}):
+        echodata = cls(
+            converted_raw_path=converted_raw_path,
+            storage_options=storage_options,
+            open_kwargs=open_kwargs,
+        )
+        echodata._check_path(converted_raw_path)
+        converted_raw_path = echodata._sanitize_path(converted_raw_path)
+        suffix = echodata._check_suffix(converted_raw_path)
+        tree = open_datatree(
+            converted_raw_path,
+            engine=XARRAY_ENGINE_MAP[suffix],
+            **echodata.open_kwargs,
+        )
+        # Change root name to top
+        tree.name = "Top-level"
+
+        echodata._set_tree(tree)
+
```

Comment:
@b-reyes I'm thinking that whatever internal restructuring method for `0.5.x` to `0.6.x` can go here. I suggest modifying the internal `self._tree` in the echodata object, since the actual `setattr` doesn't get called until later within `_load_tree`. This way you can still use the `version_info` and `group_paths`.

Reply:
So, after `echodata._set_tree(tree)` I can add/remove a child/parent from the tree? Similarly, can I rename coords/variables and add/delete them or does this need to happen after `setattr`?

Reply:
This function just puts the tree into `self._tree`. You can add/remove a child/parent from the tree, but seems like this is not implemented upstream in datatree yet. So you'd have to think about overwriting this tree after you get it to how you want it internally.